ext/yarp/extconf.rb

```diff
@@ -10,11 +10,12 @@
   # optimize across all of the source files.
   File.binwrite("yarp.c", Dir[File.expand_path("src/**/*.c", __dir__)].map { |src| File.binread(src) }.join("\n"))
 
-  # Tell Ruby where to find the headers for YARP, specify the C standard, and make
-  # sure all symbols are hidden by default.
+  # Tell Ruby where to find the headers for YARP, specify the C standard, and
+  # make sure all symbols are hidden by default.
+  find_header("yarp.h", File.expand_path("include", __dir__))
   $CFLAGS << " -I#{__dir__}/include -std=gnu99 -fvisibility=hidden"
 
-  $objs = %w[extension.o yarp.o node.o pack.o]
+  $objs = %w[api_node.o api_pack.o compile.o extension.o yarp.o]
```

Comment:
Do we need to specify this list? It seems rather error-prone

Reply:
Honestly I don't know. I _think_ so? I am really struggling to understand mkmf

Reply:
The issue is, `yarp.c` _won't_ be in `ext/yarp` because we create it immediately above. So maybe I just need `$objs << "yarp.o"`?

Reply:
Do we need this BUNDLE mode? Or could we just always use the case below?
Concatenating seems not so nice if one ever has to debug those files, and we could use some LTO flags if they prove effective.

Reply:
@kddnewton It shouldn't matter -- the code I pointed to above is in `create_makefile` and the source files are determined immediately above the code I linked.

Try removing the `$objs` line and see if it breaks? It seems to work OK for me (but my dev env may not be set up correctly).

Reply:
Oh, I see, you have source files in $srcdir *and* the build directory. :thinking:

Reply:
> So maybe I just need `$objs << "yarp.o"`?

It should work.